### PR TITLE
1188 new game UI combat figthing

### PR DESCRIPTION
**Pull request by Spintacle123**
### PR Description

PR Includes
- New UI for Combat Fighting/Adventure  w/ mobile responsive
- Character Sidebar
- also added route page for Play-to-Earn Page

### Testing

Has the code been tested, or does it need double checking?

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-      <router-view v-if="canShowApp" />
+      <b-row>
+        <b-col cols="3" lg="3" md="3" sm="2" :class="toggleSideBar ? 'sideBar triggerMargin' : 'sideBar removeMargin'">
+          <character-bar v-if="!featureFlagStakeOnly && currentCharacterId !== null" />
+        </b-col>
+        <b-col lg="9" md="9" sm="10" :class="toggleSideBar ? 'can-show-app addFlex  bg-image' : 'can-show-app removeFlex  bg-image'">
+          <router-view v-if="canShowApp" />
+        </b-col>
+        <WeaponRowGrid v-if="showWeapon" v-model="currentWeaponId" :checkForDurability="true"/>
+      </b-row>
```
Comment (Sellie96) [inline frontend/src/App.vue:13]: Some complicated css switching here can we make this a bit more simple. The Character-bar is outside the main app and is missing some checks. can we move it and some of the styling out of the app.vue

File: frontend/package-lock.json [hunk 2]
```diff
@@ -3312,16 +3312,6 @@
           "integrity": "sha512-shAmDyaQC4H92APFoIaVDHCx5bStIocgvbwQyxPRrbUY20V1EYTbSDchWbuwlMG3V17cprZhA6+78JfB+3DTPw==",
           "dev": true
```
Comment (Sellie96) [inline frontend/package-lock.json:3313]: uncomment this file please

File: frontend/src/components/CombatResults.vue [hunk 3]
```diff
@@ -1,53 +1,67 @@
...
   <div>
     <div class="tob-bg-img promotion-decoration">
-      <img class="vertical-decoration bottom" src="../assets/border-element.png">
+      <img class="vertical-decoration bottom" src="../assets/header-ads.png">
     </div>
-    <div class="float-left crypto-warrior-image">
+    <!-- <div class="float-left crypto-warrior-image">
       <img class="vertical-decoration" src="../assets/cryptowarrior.png">
-    </div>
+    </div> -->
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:8]: Clean up the code if it isn't required remove or comment a reason why it should stay

File: frontend/src/components/CombatResults.vue [hunk 4]
```diff
@@ -1,53 +1,67 @@
...
+        </b-row>
+        <b-row class="bb">
+          <b-col cols="4"  lg="5" sm="4" md="4" class="win-details text-right no-padding">
+            <h5 class="no-margin">{{$t('combatResults.enemyRolled')}} :</h5>
+          </b-col>
+          <b-col cols="4" lg="2" sm="4" md="4" class="win-details text-center no-padding">
+            <img src="../assets/arrow-right.png" alt="">
+          </b-col>
+          <b-col cols="4" lg="5" sm="4" md="4" class="win-details text-left no-padding">
+            <h5 class="no-margin">{{fightResults.enemyRoll}}</h5>
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:32]: Can we clean up some of the css here into proper classes

File: frontend/src/components/CombatResults.vue [hunk 5]
```diff
@@ -1,53 +1,67 @@
...
+          </b-col>
+        </b-row>
+      </b-container>
       <b-container v-if="fightResults.isVictory">
         <b-row>
-          <b-col class="text-left no-padding">
+          <b-col class="you-earned tex-center no-padding">
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:38]: Miss type not actually doing anything so I'm guessing dead css

File: frontend/src/components/CombatResults.vue [hunk 6]
```diff
@@ -1,53 +1,67 @@
...
+          <b-col class="earned text-center no-padding">
             <h4>
-              {{formattedUsd}}
+              {{formattedSkill}}  {{formattedUsd}}
               <Hint :text="$t('combatResults.hint')" />
             </h4>
-            <h6 class="formatted-skill">{{formattedSkill}}</h6>
-            <h5>{{formattedXpGain}}</h5>
+            <!-- <h6 class="formatted-skill"></h6> -->
+            <h5>+ {{formattedXpGain}}</h5>
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:49]: Commented with a duplicate above one needs removing entirely

File: frontend/src/components/CombatResults.vue [hunk 7]
```diff
@@ -184,9 +199,116 @@ export default Vue.extend({
...
+
+.bb{
+  border-bottom: 1px solid rgba(255, 255, 255, 0.282);
+  width: 70%;
+  margin: auto;
+}
+
+.bt{
+  border-top: 1px solid rgba(255, 255, 255, 0.282);
+
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:212]: bb?
bt? 

Can they be slightly more descriptive or added to a css class that requires them specifically

File: frontend/src/components/CombatResults.vue [hunk 8]
```diff
@@ -184,9 +199,116 @@ export default Vue.extend({
...
+  .float-center .container{
+    padding: 0px !important;;
+  }
+
+  .float-center .container .bb{
+    padding: 0px !important;
+    width: 100% !important;
+  }
+}
```
Comment (Sellie96) [inline frontend/src/components/CombatResults.vue:312]: Same as above comment just needs a tidy up and thinking about different screen sizes. The overuse of !important shows that there is conflicting css you are overriding

File: frontend/src/components/NavBar.vue [hunk 9]
```diff
@@ -28,7 +28,7 @@
         <options class="options-display-mobile"/>
       </div>
     </b-navbar>
-    <claim-rewards-bar v-if="canShowRewardsBar" />
+    <!-- <claim-rewards-bar v-if="canShowRewardsBar" /> -->
```
Comment (Sellie96) [inline frontend/src/components/NavBar.vue:31]: remove

File: frontend/src/components/WeaponInvetory.vue [hunk 10]
```diff
@@ -0,0 +1,727 @@
...
+        </div>
+    </div>
+
+    <!-- inventory selected weapon list display -->
+    <div v-if="displayType == 'inventory' || !isMobile()" class="glow-container" ref="el" :class="['glow-' + (weapon.stars || 0)]">
+      <div class="weapon-flex">
+        <div :class="'weapon-img frame-'+ (weapon.stars || 0)">
+            <!-- WEAPON ID -->
+          <div class="id">{{$t('weaponIcon.id')}} {{ weapon.id }}</div>
+          <!-- below use of weapon.id is for test purpose, should be replaced with getWeaponCosmetic(weapon.id) -->
```
Comment (Sellie96) [inline frontend/src/components/WeaponInvetory.vue:38]: Is this relevant? Also file name has a typo. Maybe tidy up the html here too. Inconsistent spacing

File: frontend/src/components/WeaponInvetory.vue [hunk 11]
```diff
@@ -0,0 +1,727 @@
...
+    position: absolute;
+    bottom:10px;
+    right:10px;
+  }
+
+  .img-adventure{
+    border-radius: 7px;
+  }
+}
+
```
Comment (Sellie96) [inline frontend/src/components/WeaponInvetory.vue:726]: This is a monstrous file. There is definitely some code that could be cleaned up / removed. but I think this was a copy and paste from somewhere else?

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 12]
```diff
@@ -1,27 +1,27 @@
...
-            :portrait="true" />
-          <span v-if="isLoadingCharacter" style="position: relative">
+            :portrait="true" /> -->
+          <!-- <span v-if="isLoadingCharacter" style="position: relative">
             <div class="loading-container">
               <i class="fas fa-spinner fa-spin"></i>
             </div>
           </span>
-        </div>
+        </div> -->
```
Comment (Sellie96) [inline frontend/src/components/smart/CharacterDisplay.vue:22]: delete commented code

**Comment by Sellie96** [comment]
Yeah there was a lot more to clean up. Can you just clean up this PR and sort the css out. Also add all the new text to the translation file instead or the raw text. Will make a big difference to reviewing

File: frontend/src/App.vue (hunk 1 above)
Comment (Spintacle123) [reply frontend/src/App.vue:13]: > Some complicated css switching here can we make this a bit more simple. The Character-bar is outside the main app and is missing some checks. can we move it and some of the styling out of the app.vue

Im not sure, what happened there, In my end here's my code.
code

Comment (Spintacle123) [reply frontend/src/App.vue:13]: > Some complicated css switching here can we make this a bit more simple. The Character-bar is outside the main app and is missing some checks. can we move it and some of the styling out of the app.vue

I will only remove/move the css styling that I added.

File: frontend/src/views/PlayToEarn.vue [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+                  <p>Join the players in defeating the boss.</p>
+                  <div class="btn-join">JOIN</div>
+                </div>
+            </router-link>
+
+            <!-- change this to arena route -->
+            <router-link v-if="!stakeOnly" :to="{ name: 'combat' }" exact class="arena navs">
+                <div>
+                  <h4>ARENA</h4>
+                  <p>Defeat others Players on PVP battles</p>
```
Comment (Sellie96) [inline frontend/src/views/PlayToEarn.vue:33]: Please add all these to the translation file

**Comment by Sellie96** [comment]
I believe a lootbox icon was removed in this PR for an extra reward on unlikely victories can that be reimplemented please

**Comment by Sellie96** [comment]
Still could do with some code cleanup too. Happy to do a cleanup commit for you if you do the rest just lemme know

**Comment by Spintacle123** [comment]
> Still could do with some code cleanup too. Happy to do a cleanup commit for you if you do the rest just lemme know

sure, it's a big help.

File: frontend/src/views/PlayToEarn.vue (hunk 13 above)
Comment (Spintacle123) [reply frontend/src/views/PlayToEarn.vue:33]: > Please add all these to the translation file

ok will do

**Comment by Spintacle123** [comment]
> I believe a lootbox icon was removed in this PR for an extra reward on unlikely victories can that be reimplemented please

currently there's hidden code for lootbox design.

File: frontend/src/views/PlayToEarn.vue (hunk 13 above)
Comment (Spintacle123) [reply frontend/src/views/PlayToEarn.vue]: done

**Comment by Spintacle123** [comment]
> image
> 
> Can you match up the Container with the logo and adjust there sizes together so it looks like a framework. Maybe reduce the size of the logo for this as it's too much of the screen currently

I already modified it.

**Comment by Spintacle123** [comment]
> I believe a lootbox icon was removed in this PR for an extra reward on unlikely victories can that be reimplemented please

I already added the lootbox design, but it's not yet functional it just ui..

**Comment by Sellie96** [comment]
image

Need to update your breakpoints again they go wonky as it gets smaller

**Comment by Sellie96** [comment]
image

Pvp clipping

**Comment by Spintacle123** [comment]
> Lootbox is implemented on live ?

I think it's not yet implemented.

**Comment by Spintacle123** [comment]
> > image
> > Need to update your breakpoints again they go wonky as it gets smaller
> 
> It has the same breakpoint, i guess the icons in the navbar pushes the div of the logo. coz it has minimum width, pushh

**Comment by Sellie96** [comment]
Sorry I should have been more clear can we append it to the right of the navbar

**Comment by Sellie96** [comment]
image

here is an example of the framework and reaction I mean

**Comment by Spintacle123** [comment]
> image
> 
> here is an example of the framework and reaction I mean

ow got it. will try

**Comment by Spintacle123** [comment]
> image
> 
> here is an example of the framework and reaction I mean

need to reconstruct the gridding system.

**Review by Sellie96 — APPROVED** [review]

**Pull request merged**